plugins/modules/aci_aaa_domain.py

```diff
@@ -0,0 +1,279 @@
+  returned: info
+  type: list
+  sample:
+    [
+        {
+            "fvTenant": {
+                "attributes": {
+                    "descr": "Production",
+                    "dn": "uni/tn-production",
+                    "name": "production",
+                    "nameAlias": "",
+                    "ownerKey": "",
+                    "ownerTag": ""
+                }
+            }
+        }
+    ]
+proposed:
+  description: The assembled configuration from the user-provided parameters
+  returned: info
+  type: dict
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment",
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(
+        name=dict(type="str", aliases=["aaa_domain"]),
+        description=dict(type="str", aliases=["descr"]),
+        restricted_rbac_domain=dict(type="bool", choices=[False, True]),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+        name_alias=dict(type="str"),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["name"]],
+            ["state", "present", ["name"]],
+        ],
+    )
+
+    name = module.params.get("name")
+    description = module.params.get("description")
+    restricted_rbac_domain = module.params.get("restricted_rbac_domain")
+    state = module.params.get("state")
+    name_alias = module.params.get("name_alias")
+
+    aci = ACIModule(module)
+    aci.construct_url(
+        root_class=dict(
+            aci_class="aaaDomain",
+            aci_rn="userext/domain-{0}".format(name),
+            module_object=name,
+            target_filter={"name": name},
+        ),
+    )
+    aci.get_existing()
+    restricted_rbac_domain_mapping = {False: "no", True: "yes"}
+    restricted_rbac_domain_state = restricted_rbac_domain_mapping.get(restricted_rbac_domain)
```

Comment:
seems like restricted_rbac_domain_state is only used for present, maybe you want to move it into if state=="present"